### PR TITLE
Add new profile functionality to rosworkon 

### PR DESCRIPTION
**Pull request by Baycken**
* Add new profile functionality to `rosworkon`
* Add explanation in README for `rosworkon` and `rosconnect`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,6 +66,32 @@ repository exists when the script runs, the robot specific setup will not run,
 and you will need to `cd` to the robot repository, optionally checkout to a
 branch, and run `./setup.sh` to setup robot specific settings.**
 
+## Functions
+### rosworkon
+To switch between project, you can use the function `rosworkon`:
```
Comment (foochuanwei) [inline README.md]: Should 'project' have an s at the end?

File: README.md [hunk 2]
```diff
@@ -66,6 +66,32 @@ repository exists when the script runs, the robot specific setup will not run,
...
+
+```bash
+rosworkon <project> [profile]
+```
+
+#### Parameters:
+- `<project>`: name of the project you wish to switch to (ex. auv). This
+parameter is required.
+- `[profile]`: catkin profile of the project you wish to switch to. This
+parameter is nor required, leave it empty if you don't know the profile to
```
Comment (foochuanwei) [inline README.md]: '... parameter is not required. Leave it ...'?

File: README.md [hunk 3]
```diff
@@ -66,6 +66,32 @@ repository exists when the script runs, the robot specific setup will not run,
...
+use.
+
+### rosconnect
+To use a remote rosmaster, you can used the function `rosconnect`:
```
Comment (foochuanwei) [inline README.md]: 'use'?

File: roboticrc [hunk 4]
```diff
@@ -28,30 +28,66 @@ function rosconnect {
...
+        unset active_profile
+      fi
+    fi
+
+    # Set TOPICS_PATH for bagging
+    if [[ -z "${PROFILE}" ]] || [[ "${PROFILE}" == "default" ]]; then
+      export TOPICS_PATH=${ROBOTIC_PATH}/${ROBOT}/topics
+    elif [[ ! -e ${profile_path}/topics ]]; then
+      echo -e "Profile selected does not have a \"topics\" file"
+      echo "Falling back to default topics..."
```
Comment (anassinator) [inline roboticrc]: this will print out all the time, so remove it

File: roboticrc [hunk 5]
```diff
@@ -28,30 +28,66 @@ function rosconnect {
...
+        pushd ${ROS_WORKSPACE}
+        catkin profile set ${PROFILE} > /dev/null
+        popd
+      fi
+    else
+      # When the second parameter is not set, find the current active profile
+      active_profile=${ROS_WORKSPACE}/.catkin_tools/profiles/profiles.yaml
+      if [[ -e ${active_profile} ]]; then
+        active_profile=$(head -n 1 ${active_profile})
+        export PROFILE=${active_profile:8}
```
Comment (anassinator) [inline roboticrc]: I think we should call it something like `CATKIN_PROFILE`

File: README.md [hunk 6]
```diff
@@ -66,6 +66,32 @@ repository exists when the script runs, the robot specific setup will not run,
...
+
+### rosconnect
+To connect to a remote ROS master, you can use the function `rosconnect`:
+
+```bash
+rosconnect [host_name]
+```
+#### Parameter:
+- `[host_name]`: the name of the host you wish to connect to, the host must be
+present in your `/etc/hosts` file. This parameter is not required, leave it
```
Comment (anassinator) [inline README.md]: required. Leave it

File: README.md [hunk 7]
```diff
@@ -66,6 +66,32 @@ repository exists when the script runs, the robot specific setup will not run,
...
+
+```bash
+rosworkon <project> [profile]
+```
+
+#### Parameters:
+- `<project>`: name of the project you wish to switch to (ex. auv). This
+parameter is required.
+- `[profile]`: catkin profile of the project you wish to switch to. This
+parameter is nor required. Leave it empty if you don't know the profile to
```
Comment (anassinator) [inline README.md]: not required

File: roboticrc (hunk 4 above)
Comment (Baycken) [inline roboticrc]: the default and empty profile will not produce this warning, we just have to renforce that each profile we create must have a `topics` file

Comment (anassinator) [inline roboticrc]: I don't like that, I think a profile without a topics file should fallback to the default topics file without warning

Comment (Baycken) [inline roboticrc]: Sure, if you feel strongly about it

File: roboticrc (hunk 5 above)
Comment (Baycken) [inline roboticrc]: do we really need this parameter to be exported? I can change it to local variable and unset it

**Pull request merged**